counterpartylib/lib/util.py

```diff
@@ -198,6 +202,79 @@ def get_asset_name (db, asset_id, block_index):
     elif not assets:
         return 0    # Strange, I know…
 
+# checks and validates subassets
+#   throws exceptions for assset or subasset names with invalid syntax
+def parse_subasset_from_asset_name(asset):
+    subasset_parent = None
+    subasset_child = None
+    subasset = None
+    chunks = asset.split('.', 1)
+    if (len(chunks) == 2):
+        subasset_parent = chunks[0]
+        subasset_child = chunks[1]
+        subasset = asset
+
+        # validate parent asset
+        validate_subasset_parent_name(subasset_parent)
+
+        # validate child asset
+        validate_subasset_name(subasset, subasset_child)
+
+    return (subasset_parent, subasset)
+
+# takes an asset description in 
+#   if it has a suffix like ;;lPARENT.child, then parse the subasset information
+#   throws exceptions for invalid asset names
+def parse_subasset_from_description(description):
+    subasset_parent = None
+    subasset_child = None
+    subasset = None
+    chunks = description.rsplit(';;l', 1)
+    if len(chunks) == 2:
+        split_description, asset = chunks
+        subasset_parent, subasset = parse_subasset_from_asset_name(asset)
```

Comment:
nit: the naming is a bit confusing, "asset" in the line above, and "subasset" end up being the same string, right? Both are the full asset name, e.g. PIZZA.DOMINOS... ? if so, can you go through and clarify the use of "subasset" vs "asset" in variable names to be a bit more consistent? (I think my nit is more with "subasset = asset" in parse_subasset_from_asset_name() then...)